.travis.yml

```diff
@@ -0,0 +1,81 @@
+    # Doctr deploy key for slaclab/timechart
+    - secure: "EUgO/bmv052rKCsZADTn875axejf7aSgS+IP498YUK/7RqPFSxT+kX2GP9yrcCVMfPokcITLcmXPgrnTjAeYmVwHTdU6V19CeZ0f6uG1CprhVlxKpdgKwI/nuS98XnXfvklxaQu4ANCI5dprSbvxDvVqV0Zkd8IaG8cru5XqkAewYk3+UJot5Q2QnpRmn30juTAD8O0sdLC4Y9EJlwaf9yj1nGvUfryxNJrDES9W/L6w/8Oez1hIQsL/1r9IeU6tksHUkfv+MC/e26yskii5Gbyu2G50w3WY5dzkQI42ggYQeI2kb3a/rka9x4wx1S8eDWpAzZkgKnpnzfJd6bsG5y4yIxfihy98mFbZwWiayFMYNdFf5nud2j8qwrrvXUmTJBQrAlpj8eSDkzoeuutdk4vS3phtUR2OELwTCSf2YEdNxIIBNu/B2l/0aBBW2sEaxIfwQAkMNXPR1KLSw5kqOIVgQ0Hef8+lcdlFE39aiDKJ+iBpgWs33z5IBSk7TC3YMyfbqm3O7YAztca/PIILT0pslsWbSBfJEKIhT/VVWwZ3wrOabuX4LSM17D5ng7dyiF4tJB1L6yJWBrBc6Qa2pOu93zM4j6YDcuenexmKazjaXFUDd9IdUbLIOlv+ZP5Qt9u+OFjpjkL47y8f48VfS+DP2E4hoMJF8AEGMO0eO1s="
+
+matrix:
+  include:
+    - python: 2.7
+      env: PYDM_CHANNEL=pydm-tag
+    - python: 3.6
+      env: PYDM_CHANNEL=pydm-tag
+    - python: 3.7
+      dist: xenial
+      env: PYDM_CHANNEL=pydm-tag
+
+install:
+  - sudo apt-get update
+  # Install windows manager
+  - sudo apt-get install -y xvfb herbstluftwm
+  # We do this conditionally because it saves us some downloading if the
+  # version is the same.
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - export PATH="$HOME/miniconda/bin:$PATH"
+  - hash -r
+  - conda config --set always_yes yes --set changeps1 no
+  - conda install conda-build anaconda-client
+  - conda update -q conda conda-build
+  - conda config --add channels $PYDM_CHANNEL
+  - conda config --append channels conda-forge
+  # Useful for debugging any issues with conda
+  - conda info -a
+  # Grab all dependencies
+  - conda build -q conda-recipe --python=$TRAVIS_PYTHON_VERSION --output-folder bld-dir
+  - conda config --add channels "file://`pwd`/bld-dir"
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION timechart --file dev-requirements.txt
+  # Launch Conda environment
+  - source activate test-environment
+
+before_script:
+  # Taken from docs.travis-ci.com/user/gui-and-headless-browsers
+  - "export DISPLAY=:99.0"
+  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1920x1200x24 -ac +extension GLX +render -noreset"
+  - sleep 3 # give xvfb some time to start
+  # Run windows manager
+  - "herbstluftwm &"
+  - sleep 1
+
+script:
+  - coverage run run_tests.py --disable-warnings
+  - coverage report -m
+
+  # Build docs
+  - set -e
+  - |
+    if [[ -n "$DOCTR_DEPLOY_ENCRYPTION_KEY" && $BUILD ]]; then
+      # Create HTML documentation
+      pushd docs
+      make html
+      popd
+      #Publish docs.
+      doctr deploy . --built-docs docs/build/html --deploy-branch-name gh-pages
+    fi
+after_success:
+  - codecov
+  - |
+    if [[ $TRAVIS_PULL_REQUEST == false && $TRAVIS_REPO_SLUG == $OFFICIAL_REPO && $TRAVIS_BRANCH == $TRAVIS_TAG  && $TRAVIS_TAG != '' && $BUILD ]]; then
```

Comment:
This going to go to the `pydm` channels?

Reply:
No, because the conda token isn't set up. This will not take an effect for now. After the token is set, we'll have the upload take place.